### PR TITLE
Easier portrait variants

### PR DESCRIPTION
**Pull request by CelticMinstrel**
It's common in campaigns for major characters to have several different portraits, eg for different emotions. Often these are named things like "leader.png", "leader-mad.png", etc. This is an idea to make it simpler to select a portrait variant without needing to repeat the entire file path.

Here's how it works: Suppose the base portrait is "leader.png", and you want the "mad" variant which is at "leader-mad.png". You write your message tag like this:

```wml
[message]
   speaker=leader
   image_variant=mad
   message=_"I'm really mad right now!"
[/message]
```

And the `[message]` tag automatically inserts the string "-mad" right before the file extension. (It currently assumes the file extension is ".png".)

But what if you have a similar situation, but the base portrait is called "leader-neutral.png"? This is easy too. Instead of setting `profile=leader-neutral.png`, you set `profile=leader-%neutral%.png`, and write your message tag just like above. The string between the two percent signs will be replaced with the image variant, resulting in "leader-mad.png". If no image variant is specified, the percent signs are removed.

### CODE REVIEW

**Comment by Elvish-Hunter** [comment]
My main concern with this feature is how `wmlscope` is supposed to handle it. Bear in mid that wmlscope doesn't have a proper parser, so when generating the cross-reference file list it just iterates a file line by line, matching it with a regex to collect possible file names.
Not only that: let's admit for a moment that wmlscope (although it's really wmltools' `CrossRef` class) might be fixed to parse `[message]` tags to generate file names; if `speaker=` has a sensible value, like "Konrad", everything should be fine. But if speaker is "unit", wmlscope will have to search for a "unit-mad.png" file? Or check if every available portrait has a "mad" variation?
In other words, I feel like this feature will spare using some macros, but on the other hand it'll add maintenance issues with no easy solutions.

**Comment by CelticMinstrel** [comment]
Hmm. You've actually come up with something that can be done with this but not with macros... though, the way I did this, it's not flexible enough to be used with `speaker=unit` since it wouldn't fall back if the speaker _didn't_ have the required variation.

I'm not sure I fully understand the issue, but that's probably because I don't understand what `wmlscope` is supposed to do (cross-reference file list?). I do get the general idea of what the problem is. Even though you've come up with a use-case that can't be done with macros, I don't think it's a very useful use-case (and in the limited cases where it _could_ be useful – in conjunction with a filter that can match only a few units – you can still manage to macroize it). This was something I whipped up quickly to see what people thought, so I'm not particularly attached to it.

**Comment by CelticMinstrel** [comment]
Closing because a) it was a quick thing I whipped up in an hour or two, b) there have been decent arguments against it, and c) it's cluttering up the PR list.
